Source/items.cpp

```diff
@@ -4656,36 +4482,32 @@ static void SpawnOnePremium(int i, int plvl, int myplr)
 		switch (items[0]._itype) {
 		case ITYPE_LARMOR:
 		case ITYPE_MARMOR:
-		case ITYPE_HARMOR:
-			ivalue = get_armor_max_value(myplr);
+		case ITYPE_HARMOR: {
+			auto mostValuablePlayerArmor = plr[myplr].GetMostValuableItem(
+			    [](const ItemStruct &item) {
+				    return item._itype == ITYPE_LARMOR
+				        || item._itype == ITYPE_MARMOR
+				        || item._itype == ITYPE_HARMOR;
+			    });
+
+			ivalue = mostValuablePlayerArmor == nullptr ? 0 : mostValuablePlayerArmor->_iIvalue;
 			break;
+		}
 		case ITYPE_SHIELD:
-			ivalue = get_shield_max_value(myplr);
-			break;
 		case ITYPE_AXE:
-			ivalue = get_axe_max_value(myplr);
-			break;
 		case ITYPE_BOW:
-			ivalue = get_bow_max_value(myplr);
-			break;
 		case ITYPE_MACE:
-			ivalue = get_mace_max_value(myplr);
-			break;
 		case ITYPE_SWORD:
-			ivalue = get_sword_max_value(myplr);
-			break;
 		case ITYPE_HELM:
-			ivalue = get_helm_max_value(myplr);
-			break;
 		case ITYPE_STAFF:
-			ivalue = get_staff_max_value(myplr);
-			break;
 		case ITYPE_RING:
-			ivalue = get_ring_max_value(myplr);
-			break;
-		case ITYPE_AMULET:
-			ivalue = get_amulet_max_value(myplr);
+		case ITYPE_AMULET: {
+			auto mostValuablePlayerItem = plr[myplr].GetMostValuableItem(
+			    [](const ItemStruct &item) { return item._itype == items[0]._itype; });

```

Comment:
1. This is different from the block below which uses `itemType`
2. This lambda does not capture `items` so it does not have access to it

Reply:
items is the global list of items in a level

Reply:
> 1. This is different from the block below which uses itemType

Yes it is. They were different already before this change.